contracts/badger-sett/strategies/unitprotocol/StrategyUnitProtocolMeta.sol

```diff
@@ -0,0 +1,254 @@
+
+abstract contract StrategyUnitProtocolMeta is BaseStrategy {
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+    using AddressUpgradeable for address;
+    using SafeMathUpgradeable for uint256;
+
+    // Unit Protocol module: https://github.com/unitprotocol/core/blob/master/CONTRACTS.md
+    address public constant cdpMgr01 = 0x0e13ab042eC5AB9Fc6F43979406088B9028F66fA;
+    address public constant unitVault = 0xb1cFF81b9305166ff1EFc49A129ad2AfCd7BCf19;
+    address public constant unitVaultParameters = 0xB46F8CF42e504Efe8BEf895f848741daA55e9f1D;
+    address public constant debtToken = 0x1456688345527bE1f37E9e627DA0837D6f08C925;
+    address public constant eth_usd = 0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419;
+
+    // sub-strategy related constants
+    address public collateral;
+    uint256 public collateralDecimal = 1e18;
+    address public unitOracle;
+    uint256 public collateralPriceDecimal = 1;
+    bool public collateralPriceEth = false;
+
+    // configurable minimum collateralization percent this strategy would hold for CDP
+    uint256 public minRatio = 200;
+    // collateralization percent buffer in CDP debt actions
+    uint256 public ratioBuff = 200;
+    uint256 public constant ratioBuffMax = 10000;
+
+    // **** Modifiers **** //
+
+    function _onlyCDPInUse() internal view {
+        uint256 collateralAmt = getCollateralBalance();
+        require(collateralAmt > 0, "!zeroCollateral");
+
+        uint256 debtAmt = getDebtBalance();
+        require(debtAmt > 0, "!zeroDebt");
+    }
+
+    // **** Getters ****
+
+    function getCollateralBalance() public view returns (uint256) {
+        return IUnitVault(unitVault).collaterals(collateral, address(this));
+    }
+
+    function getDebtBalance() public view returns (uint256) {
+        return IUnitVault(unitVault).getTotalDebt(collateral, address(this));
+    }
+
+    function getDebtWithoutFee() public view returns (uint256) {
+        return IUnitVault(unitVault).debts(collateral, address(this));
+    }
+
+    function debtLimit() public view returns (uint256) {
+        return IUnitVaultParameters(unitVaultParameters).tokenDebtLimit(collateral);
+    }
+
+    function debtUsed() public view returns (uint256) {
+        return IUnitVault(unitVault).tokenDebts(collateral);
+    }
+
+    /// @dev Balance of want currently held in strategy positions
+    function balanceOfPool() public override view returns (uint256) {
+        return getCollateralBalance();
+    }
+
+    function collateralValue(uint256 collateralAmt) public view returns (uint256) {
+        uint256 collateralPrice = getLatestCollateralPrice();
+        return collateralAmt.mul(collateralPrice).mul(1e18).div(collateralDecimal).div(collateralPriceDecimal); // debtToken in 1e18 decimal
+    }
+
+    function currentRatio() public view returns (uint256) {
+        _onlyCDPInUse();
+        uint256 collateralAmt = collateralValue(getCollateralBalance()).mul(100);
```

Comment:
Wouldn't you want to convert both collateral and debt, or convert neither of them since (x * c) / (y * c= x / y ?